### ISSUE TITLE
Allow disabling builtin extensions from the command line

### ISSUE THREAD

**Issue by vicapow** [Microsoft/vscode#52639]
Allow disabling builtin extensions from the command line

**Comment by sandy081** [comment]
@vicapow Do you want to disable all built in extensions or only specific extension?

**Comment by vicapow** [comment]
Either would work for our use case. I saw there’s a disable all extensions but that doesn’t actually turn off built in extensions

**Comment by vicapow** [comment]
we Just have some extensions we need to disable but disabling all would achieve the former.

The use case is to automatically provision laptops with vscode set up to use Flowtype and have Typescript disabled. Having both enabled is confusing

**Comment by vicapow** [comment]
Any update on this? Is there interest in taking this as a PR?

**Comment by sandy081** [comment]
@vicapow VS Code has support now to disable extensions from command line. For eg:

`code --disable-extension vscode.typescript-language-features`

Let me know if this is sufficient for you to build the support you need

**Comment by sandy081** [comment]
It should be available in 1.26.1 

May I know What is not working?

**Comment by vicapow** [comment]
This doesn't seem to persist the extension disablement. It only opens the editor with those extensions disabled.

**Comment by sandy081** [comment]
@vicapow Yes, it does not persist as these are the options given to that window. But this will be carried for all windows you open from that window.

**Comment by vicapow** [comment]
Okay, thanks for clarify. With that said, this still doesn't solve for our use case. I've updated the title.

**Comment by sandy081** [comment]
@vicapow 

> The use case is to automatically provision laptops with vscode set up to use Flowtype and have Typescript disabled. Having both enabled is confusing

What is the approach you are expecting here? How do you want to provision VS Code with such a set up? Is not it possible for you in the provisioned laptops that will always run code with --disable-extension `vscode.typescript-language-features` ?

**Comment by vicapow** [comment]
Specifically I would like to avoid having to wrap the `code` command if possible. I would expect this to be somewhat error prone and confusing to debug as an end user.

**Comment by sandy081** [comment]
@vicapow So you need some kind of UI action in VS Code?

**Comment by vicapow** [comment]
@sandy081 no just some way to persist the state that the built in plugin has been disabled beyond subsequent invocations and instantiations of the editor.

**Comment by sandy081** [comment]
@vicapow For persisting the state user has to invoke the action some how either through cli or from UI. If that's the case why not open the UI and disable TS extension completely?

**Comment by vicapow** [comment]
> For persisting the state user has to invoke the action some how either through cli or from UI. 

The user isn't the one initiation the action, it's our provisioning scripts that run automatically on each users laptop.

We could wrap the `code` CLI to always run with the proper flags but then if someone opened the app directly, the extensions would no longer be disabled which is inconsistent.

> If that's the case why not open the UI and disable TS extension completely?
We could ask users to do this but it's a manual step that can ideally be avoided.

**Comment by vicapow** [comment]
That’s okay. We don’t want to prevent them from using Typescript just make the standard at our company the default.

Once disabled they can always renable it but this would be the exception. Most projects are using Flow and  it wouldn’t make sense to have both  flow and TS extensions enabled by default.

**Comment by sandy081** [comment]
I see. Then I do not think this is a requirement to disable an extension by default when VS Code is installed, instead a way to have a project specific disabled extensions. When ever user opens that project, these extensions has to be disabled.

**Comment by vicapow** [comment]
That would help but we're looking to have these default settings apply to all projects, not just one in particular.

**Comment by lxe** [comment]
I think the crux of the problem is this:

```
https://github.com/Microsoft/vscode/issues/15611#issuecomment-349231959
```

> Enabled/Disabled extensions are not stored in workspace settings. Instead it is stored in local storage cache. You can manage them from the extensions view.

It would be nice to have extension disabled/enabled state to be managed from the config file (per workspace via `.vscode/settings.json`, or per user via the [user config](https://code.visualstudio.com/docs/getstarted/settings#_settings-file-locations))

Also see https://github.com/Microsoft/vscode/issues/15611#issuecomment-287262985

**Comment by vicapow** [comment]
I don't have much of a preference on how to achieve this but somehow, we would like this state to persist in a way we can provision these machines to default to disable TS builtin extensions for all projects. The idea above was to have some sort of configurable JSON settings somewhere in `~/` that gets read by VSCode so it knows which extensions to disable by default when the app opens.

**Comment by sandy081** [comment]
I see. I think we can enhance current support of disabling extension from command line to persist the state but it needs new storage support - https://github.com/Microsoft/vscode/issues/58957

**Comment by Rendez** [comment]
After reading the thread in detail I gathered that we can only open a workspace in VSCode with some disabled built-ins via CLI. I also had the extra challenge of opening a workspace using `remote-ssh` by default, without having to find recently opened workspaces or projects in the context menu and then open that window that is actually connected to a remote container. Both things can be accomplish from CLI, but it's inconvenient to remember the exact command+arguments, so for our devs I created a Mac App (for instance called IDE) that they can either double click or open with `open .vscode/IDE.app`.

Inside the App, there's a simple bash file:

```
#!/usr/bin/env bash

/usr/local/bin/code --folder-uri "vscode-remote://ssh-remote%2B__YOUR_REMOTE_SSH_/some/root/folder/" --disable-extension vscode.github --disable-extension vscode.github-authentication --disable-extension vscode.git --disable-extension vscode.jake --disable-extension ms-vscode.node-debug --disable-extension vscode.php-language-features
```

There are several ways to create a [minimal Mac app](https://github.com/thedzy/run-your-script-as-an-application). I've also added the VSCode .icns icon file so in the end I have this:

![image](https://user-images.githubusercontent.com/61776/87053181-f33f7700-c201-11ea-9c6f-bbe9ede9362f.png)

Drag that app to your Dock, opens with the remote-ssh container + some built-ins disabled.
